unitTests/ui/RegisteredNamedMethodsTest.ts

```diff
@@ -286,6 +291,18 @@ export function RegisteredNamedMethodsTest() {
         expect(setup.cmp.client instanceof NoopAnalyticsClient).not.toBeTruthy();
       });
 
+      it('should update the search interface analytics client after being re-enabled', () => {
+        RegisteredNamedMethod.disableAnalytics(setup.env.root);
+        RegisteredNamedMethod.enableAnalytics(setup.env.root);
+        expect(setup.cmp.client).toBe(setup.env.searchInterface.usageAnalytics);
+      });
+
+      it('should cancel pending events when disabling analytics', () => {
+        const cancelAllPendingEventsSpy = spyOn(setup.cmp.client, 'cancelAllPendingEvents');
+        RegisteredNamedMethod.disableAnalytics(setup.env.root);
+        expect(cancelAllPendingEventsSpy).toHaveBeenCalled();
+      });
+
```

Comment:
A test we could add is to do a "real" simulation with a simple component. 

It would not be a "unit test" in the sense that it is testing one thing in isolation, but that the whole feature work properly as a whole.

For example, we could have a test where we setup a search interface, with an actual analytics component, and a component (eg: searchbox).

Call searchbox.submit(), check that analytics tried to send a network call/that the live analytics client method was called.

After that, disable analytics.

Then, recall searchbox.submit()

Check that there was no network call/that it is now calling noop analytics method instead.

I'm not 100% sure what the test would look like in the end, or how complicated it would be, but otherwise it seems we aren't really testing that it "works".

Reply:
that absolutely possible